proposals/nnnn-multiple-transports.md

```diff
@@ -0,0 +1,262 @@
+ 
+     :
+ 
+-    <param name="deviceInfo" type="Common.DeviceInfo" mandatory="true">
++    <param name="deviceInfo" type="Common.DeviceInfo" minsize="1" maxsize="10" array="true" mandatory="true">
+        <description>The ID, serial number, transport type the named-app's-device is connected over to HU.</description>
+     </param>
+```
+
+## Potential downsides
+
+This feature introduces additional logics in both Core and Proxy. They may increase CPU usage on a head unit and mobile phone.
+
+This feature requires Core and Proxies to buffer sent messages for a while. This will increase memory usage.
+
+When a single device is connected to SDL Core through multiple transports, different values of `DeviceUID` will be assigned par transport. Also, HMI receives multiple `DeviceInfo` information, although there is actually one device. (Note that these behaviors are same as of now and it is not a new issue.) The cause of these issues is that the device information is strictly tied to TransportAdapter classes in Core, and we have no way to merge multiple device information into one. It may cause an issue if HMI implements a "Device chooser" GUI based on information from `BasicCommunication.OnUpdateDeviceList` notification.
+
+## Impact on existing code
+
+* Most of the code changes are limited to transport and protocol handler layers, notably:
+  - Addition of new selector classes in Proxies and Core
+  - Modification to protocol handlers to support new parameters in Version Negotiation
+  - Additional implementation in protocol handlers to support Receiver Report frame
+* SDL Core's Application Manager also needs updates on app\_id usage (i.e. replacing connection\_key with App Instance ID).
+* This proposal also introduces small update on Proxies' APIs.
+* This proposal includes HMI\_API.xml changes that are not backward compatible. When Core is updated, HMI should be updated accordingly.
+
+## Out of scope of this proposal
+
+For better user experience, SDL Proxies need a means to automatically detect Core's IP address and TCP port number. Otherwise, a user needs to enter them manually on every SDL apps s/he is going to use. Such mechanism is not covered by this proposal and will be discussed in another proposal [SDL-nnnn: Add Service Discovery mechanism for TCP transport][service_discovery].
+
+An advanced feature which allows usage of Wi-Fi transport only to specific apps is proposed along with this multiple-transports idea. It will be discussed in [SDL-nnnn: Add capability to disable or reject an app based on app type, transport type and OS type][reg_limitation].
+
+## Alternatives considered
+
+- Introduce transport switching mechanism while keeping current "single transport" design. This should have less impacts on existing implementation since we only keep one transport between Proxy and Core at a same time. However, it will not resolve the limitation raised during SDLC workshop to utilize Wi-Fi transport.

```

Comment:
Please update second sentence to "This should have less impact on existing implementation since we only keep one transport between Proxy and Core at the same time."